src/services/pathfinding.service.ts

```diff
@@ -63,54 +86,54 @@ class PathFindingService {
    *
    * @returns The list of paths to travese between the start and end POI's
    */
-  private findBuildingPath = (start: POI, end: POI): FloorPath[] | null => {
-    // Get travel nodes for the floors of the start and end of path
-    const startFloorNodes = buildingFloors.find(
-      (floor) =>
-        floor.buildingId === start.buildingId && floor.level === start.level
-    ).travelNodes;
-    const endFloorNodes = buildingFloors.find(
-      (floor) =>
-        floor.buildingId === end.buildingId && floor.level === end.level
-    ).travelNodes;
-
+  private findBuildingPath = (
+    start: POI,
+    end: POI,
+    accessibilityMode: boolean
+  ): FloorPath[] | null => {
     if (start.level === end.level) {
       return [
         {
-          path: this.findPathOnFloor(
-            startFloorNodes,
-            start.location,
-            end.location
-          ),
+          path: this.findPathOnFloor(start, end, accessibilityMode),
           buildingId: start.buildingId,
           level: start.level,
         },
       ];
     }
 
-    const escalatorDirection =
-      start.level < end.level
-        ? POICategory.EscalatorUp
-        : POICategory.EscalatorDown;
+    let POIcategories;
```

Comment:
Can you define its type? Also can you rename the variable to something more descriptive because I noticed the value of this variable depends on the accessibilityMode but I didn't expect that based on the name.